R/utils.R

```diff
@@ -29,11 +29,65 @@ pandoc_html_simplify <- function(
   ),
   to = "markdown_strict-raw_html+pipe_tables+backtick_code_blocks"
 ) {
+  # Remove base64 embedded images to avoid bloating the output
+  html <- remove_base64_images(html)
+
   tmp_input <- withr::local_tempfile()
   writeLines(html, tmp_input)
   pandoc_convert(tmp_input, from = "html", to = to, ...)
 }
 
+#' Remove base64 embedded images from HTML
+#'
+#' Replaces <img> tags with base64 data URIs with a text placeholder
+#' containing the alt text if available.
+#'
+#' @param html Character vector of HTML content
+#' @return Character vector with images removed/replaced
+#' @noRd
+remove_base64_images <- function(html) {
+  html_text <- paste(html, collapse = "\n")
+
+  doc <- tryCatch(
+    xml2::read_html(html_text),
+    error = function(e) NULL
+  )
+
+  if (is.null(doc)) {
+    return(html)
+  }
+
+  img_nodes <- xml2::xml_find_all(doc, "//img[@src]")
+
+  if (length(img_nodes) == 0) {
+    return(html)
+  }
+
+  # Replace base64 images with text placeholders
+  for (img in img_nodes) {
+    src <- xml2::xml_attr(img, "src")
+
+    if (is.na(src) || !grepl("^data:image", src)) {
+      next
+    }
+
+    alt_text <- xml2::xml_attr(img, "alt")
+    replacement <- if (!is.na(alt_text) && nzchar(alt_text)) {
+      paste0("[Image: ", alt_text, "]")
+    } else {
+      "[Image]"
+    }
+
+    xml2::xml_replace(
+      img,
+      xml2::read_xml(paste0("<span>", replacement, "</span>"))
+    )
```

Comment:
It turns out that `xml_replace()` supports a slightly more convenient syntax

```suggestion
    xml2::xml_replace(img, "span", replacement)
```